src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,349 @@
+                }
+            } catch (ex: Exception) {
+                logger.error("Callback URL was not properly formatted")
+            }
+        }
+        val logEntitySetId = createLogEntitySet(integrationName, integration)
+        integration.logEntitySetId = Optional.of(logEntitySetId)
+        integrations[integrationName] = integration
+        return integration.key!!
+    }
+
+    fun readIntegrationDefinition(integrationName: String): Integration {
+        checkIntegrationExists(integrationName)
+        return integrations.getValue(integrationName)
+    }
+
+    fun updateIntegrationDefinition(integrationName: String, integrationUpdate: IntegrationUpdate) {
+        checkIntegrationExists(integrationName)
+        integrations.executeOnKey(integrationName, UpdateIntegrationEntryProcessor(integrationUpdate))
+    }
+
+    fun deleteIntegrationDefinition(integrationName: String) {
+        checkIntegrationExists(integrationName)
+        val integration = integrations.getValue(integrationName)
+        integrations.remove(integrationName)
+        entitySetManager.deleteEntitySet(integration.logEntitySetId.get())
+    }
+
+    private fun checkIntegrationJobExists(jobId: UUID) {
+        checkState(integrationJobs.containsKey(jobId), "Job Id $jobId is not assigned to an existing integration job")
+    }
+
+    private fun checkIntegrationExists(integrationName: String) {
+        checkState(integrations.containsKey(integrationName), "Integration with name $integrationName does not exist.")
+    }
+
+    private fun getIdToken(creds: Properties): String {
+        return authAPI
+                .login(creds.getProperty("email"), creds.getProperty("password"), creds.getProperty("realm"))
+                .setScope(creds.getProperty("scope"))
+                .setAudience(creds.getProperty("audience"))
+                .execute()
+                .idToken
+    }
+
+    private fun getSrcDataSource(source: Properties): HikariDataSource {
+        return HikariDataSource(HikariConfig(source))
+    }
+
+    private fun generateDestinationsMap(integration: Integration, missionParameters: MissionParameters, dataIntegrationApi: DataIntegrationApi): Map<StorageDestination, IntegrationDestination> {
+        val s3BucketUrl = integration.s3bucket
+        val dstDataSource = HikariDataSource(HikariConfig(missionParameters.postgres.config))
+        integration.maxConnections.ifPresent { dstDataSource.maximumPoolSize = it }
+
+        val pgDestination = PostgresDestination(
+                entitySets.mapKeys { it.value.id },
+                entityTypes,
+                propertyTypes.mapKeys { it.value.id },
+                dstDataSource
+        )
+
+        if (s3BucketUrl.isBlank()) {
+            return mapOf(StorageDestination.POSTGRES to pgDestination)
+        }
+        val s3Api = Retrofit.Builder()
+                .baseUrl(s3BucketUrl)
+                .addConverterFactory(RhizomeByteConverterFactory())
```

Comment:
I think you can use some of the RetrofitFactory helpers to simplify this a bit, like `decorateWithFactories`